presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/MapSqlFunctions.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar.sql;
+
+import com.facebook.presto.spi.function.Description;
+import com.facebook.presto.spi.function.SqlInvokedScalarFunction;
+import com.facebook.presto.spi.function.SqlParameter;
+import com.facebook.presto.spi.function.SqlParameters;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.function.TypeParameter;
+
+public class MapSqlFunctions
+{
+    private MapSqlFunctions() {}
+
+    @SqlInvokedScalarFunction(value = "map_top_n_values", deterministic = true, calledOnNullInput = false)
+    @Description("Returns the top N values of the given map in descending order according to the natural ordering of its values.")
+    @TypeParameter("K")
+    @TypeParameter("V")
+    @SqlParameters({@SqlParameter(name = "input", type = "map(K, V)"), @SqlParameter(name = "n", type = "bigint")})
+    @SqlType("array<V>")
+    public static String mapTopNValues()
+    {
+        return "RETURN IF(n < 0, fail('n must be greater than or equal to 0'), slice(reverse(filter(array_sort(map_values(input)), e -> e IS NOT NULL)), 1, n))";
```

Comment:
Why is the NOT NULL check here? We should simply sort and return the last n elements from the values

Reply:
Cause unlike the keys, values can have nulls. And wanted to return only non nulls. So if a map has all null values then this will return empty list instead of list of nulls.

Reply:
No you can't do such things - semantics again. Just keep it clean

Reply:
Sure! but the thing is when we sort the values using array_sort it put all the null at the end and when we reverse it to get the top values I will have to offset the index to make sure we get the non null values.

For example [null, null, 4, 3, 2, 1] -> the top 2 values would be 4, 3

Now for the same array, if we want to get the top 5 values then should we return [4, 3, 2, 1] or [4, 3, 2, 1, null]?